### PR TITLE
Add descriptions to assertions in testMetrics

### PR DESCRIPTION
**Pull request by DaveCTurner**
In #101608 we saw one of these assertions fail, but it's impossible to
know which one without some more details. This commit adds descriptions
to the assertions in the loop.

### CODE REVIEW

File: modules/repository-s3/src/internalClusterTest/java/org/elasticsearch/repositories/s3/S3BlobStoreRepositoryTests.java [hunk 1]
```diff
@@ -281,8 +281,8 @@ public void testMetrics() throws Exception {
                     operation,
                     OperationPurpose.parse((String) metric.attributes().get("purpose"))
                 );
-                assertThat(statsCollectors, hasKey(statsKey));
-                assertThat(metric.getLong(), equalTo(statsCollectors.get(statsKey).counter.sum()));
+                assertThat(nodeName + "/" + statsKey, statsCollectors, hasKey(statsKey));
+                assertThat(nodeName + "/" + statsKey, metric.getLong(), equalTo(statsCollectors.get(statsKey).counter.sum()));
```
Comment (kingherc) [inline modules/repository-s3/src/internalClusterTest/java/org/elasticsearch/repositories/s3/S3BlobStoreRepositoryTests.java:285]:
```suggestion
                assertThat(nodeName + "/" + statsKey + " has correct sum", metric.getLong(), equalTo(statsCollectors.get(statsKey).counter.sum()));
```

File: modules/repository-s3/src/internalClusterTest/java/org/elasticsearch/repositories/s3/S3BlobStoreRepositoryTests.java [hunk 2]
```diff
@@ -281,8 +281,8 @@ public void testMetrics() throws Exception {
                     operation,
                     OperationPurpose.parse((String) metric.attributes().get("purpose"))
                 );
-                assertThat(statsCollectors, hasKey(statsKey));
-                assertThat(metric.getLong(), equalTo(statsCollectors.get(statsKey).counter.sum()));
+                assertThat(nodeName + "/" + statsKey, statsCollectors, hasKey(statsKey));
```
Comment (kingherc) [inline modules/repository-s3/src/internalClusterTest/java/org/elasticsearch/repositories/s3/S3BlobStoreRepositoryTests.java:284]:
```suggestion
                assertThat(nodeName + "/" + statsKey + " exists", statsCollectors, hasKey(statsKey));
```

**Review by kingherc — APPROVED** [review]
LGTM, only nits for the assert descriptions

**Comment by DaveCTurner** [comment]
I left that out since you can determine which assertion tripped from the line numbers. But happy to add them too.

**Pull request merged**